app/routes/exercises.rb

```diff
@@ -66,6 +66,28 @@ class Exercises < Core
         redirect '/dashboard'
       end
 
+      post '/exercises/:key/request-for-help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may request help."
+          redirect '/'
+        end
+        exercise.request_help!
+        flash[:success] = "You have requested help on this submission."
+        redirect "/submissions/%s" % exercise.submissions.last.key
+      end
+
+      delete '/exercises/:key/request-for-help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may cancel a request for help."
+          redirect '/'
+        end
```

Comment:
@Insti I have look into above link. It shows the methods I have added are covered by tests. I can see all my changes in the new commit. I am not sure what else needs to be done. I will check on this and get back to you!